plutus-benchmark/nofib/bench/Shared.hs

```diff
@@ -33,33 +33,30 @@ mkBenchMarks (benchClausify, benchKnights, benchPrime, benchQueens) = [
                       , bench "formula4" $ benchClausify Clausify.F4
                       , bench "formula5" $ benchClausify Clausify.F5
                       ]
-  , bgroup "knights" [ bench "4x4" $ benchKnights 100 4
+  , bgroup "knights" [ bench "2x2" $ benchKnights 100 2
+                     , bench "4x4" $ benchKnights 100 4
                      , bench "6x6" $ benchKnights 100 6
-                     , bench "8x8" $ benchKnights 100 8
                      ]
   , bgroup "primetest" [ bench "05digits" $ benchPrime Prime.P5
                        , bench "08digits" $ benchPrime Prime.P8
                        , bench "10digits" $ benchPrime Prime.P10
                        , bench "20digits" $ benchPrime Prime.P20
-                       , bench "30digits" $ benchPrime Prime.P30
-                       , bench "40digits" $ benchPrime Prime.P40
-                       , bench "50digits" $ benchPrime Prime.P50
                        -- Larger primes are available in Primes.hs, but may take a long time.
                        ]
+  , bgroup "queens3x3" [ -- N-queens problem on a 3x3 board
```

Comment:
I don't think using 3x3 instead of 5x5 will speed things up.  Criterion works by executing your program repeatedly until a time limit (by [default](https://github.com/haskell/criterion/blob/4ac74c9810790e3e83e38da4e7ff3bda15b55501/Criterion/Main/Options.hs#L82) 5 seconds) is exceeded and then it analyses the times that it's collected, so speeding up the program will  just mean that it'll get run more often but the total time will remain the same.  In fact, the time limit for `nofib` is set to [60 seconds](https://github.com/IntersectMBO/plutus/blob/bfac69f8681376b6f7e34df461efec0326b0363a/plutus-benchmark/nofib/bench/Shared.hs#L109) in order to make sure that the examples get run lots of times.  The longest-running nofib example in a [recent run](https://github.com/IntersectMBO/plutus/pull/6301#issuecomment-2222316780) of the nofib benchmarks was `queens5x5/fc`, which ran for 178ms, so it'll have been run about 337 times.  A better way to speed things up might be to leave the benchmarks unaltered but turn the limit down to 30s, or even 20s: 20s would make the entire suite run in one third of the time and still give you over 100 runs of the the slowest example, which would probably be enough to still get accurate results.

Note also that the time limits for `lists` is [15 seconds](https://github.com/IntersectMBO/plutus/blob/bfac69f8681376b6f7e34df461efec0326b0363a/plutus-benchmark/lists/bench/Bench.hs#L56) and for `validation` it's [20 seconds](https://github.com/IntersectMBO/plutus/blob/bfac69f8681376b6f7e34df461efec0326b0363a/plutus-benchmark/validation/bench/Common.hs#L107), so there may be scope for speeding things up a bit there as well.  If you alter the time limits there's maybe a danger that the results would become less reliable because we'd get fewer samples, so I'd recommend running the benchmarks a few times to check whether they become more variable: using `/benchmark` in this branch would be a good way to do that because in the base branch the benchmarks would still be run with the old limits.